processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/DefaultTaskExtractor.java

```diff
@@ -43,17 +45,33 @@ public DecatonTask<T> extract(ConsumedRecord record) {
                     taskDeserializer.deserialize(taskDataBytes),
                     taskDataBytes);
         } else {
-            try {
-                DecatonTaskRequest taskRequest = DecatonTaskRequest.parseFrom(record.value());
-                TaskMetadata metadata = TaskMetadata.fromProto(taskRequest.getMetadata());
-                byte[] taskDataBytes = taskRequest.getSerializedTask().toByteArray();
+            // There are two cases where task metadata header is missing:
+            // 1. The task is produced by an old producer which wraps tasks in DecatonTaskRequest proto.
+            // 2. The task is produced by non-DecatonClient producer.
+            //
+            // From Decaton perspective, there is no way to distinguish between these two cases,
+            // so we need to rely on a configuration to determine how to deserialize the task.
+            if (legacyFallbackEnabledProperty.value()) {
```

Comment:
h-m should this be dynamically reloadable? i just sense that it might be better fixed for the whole lifetime of this subscription, but idk

Reply:
IMO we should make property reloadable unless it's difficult or there is big downside, because it can reduce the number of rolling-restart which is a certain burden and enables quick rollback.

Since upgrade-procedure might be tough already, making reloadable might be better.

The downside I can imagine is the blast radius when property change caused a problem (because property change will happen on all instances at the same time), but quick-rollback by dynamic-reloading might be more preferred than rolling restart the application I guess.